rclrs/src/parameter/value.rs

```diff
@@ -0,0 +1,173 @@
+use crate::rcl_bindings::*;
+use std::ffi::CStr;
+
+/// A parameter value.
+///
+/// Such a value can be specified in YAML format on the command line, or in a parameter file.
+/// For instance `--param foo:='[1, 2, 3]'` specfies an `IntegerArray` value for the `foo` parameter.
+#[derive(Clone, Debug, PartialEq)]
+pub enum ParameterValue {
+    /// A boolean value.
+    ///
+    /// YAML examples: `true`, `True`, `yes`, `y`.
+    Bool(bool),
+    /// An i64 value.
+    ///
+    /// YAML examples: `1`, `-30`, `0x7C`.
+    Integer(i64),
+    /// An f64 value.
+    ///
+    /// YAML examples: `2.0`, `8e-3`.
+    Double(f64),
+    /// A string.
+    ///
+    /// YAML examples: `""`, `"björk"`, `"42"`.
+    ///
+    /// Unquoted strings are also possible, but not recommended,
+    /// because they may be interpreted as another data type.
+    String(String),
+    /// An array of u8.
+    ///
+    /// YAML example: Not possible to specify as YAML.
```

Comment:
Correct me if I'm wrong with this comment.

Reply:
Do you mean bytes (i.e. binary data) or unsigned integers? The former needs to be encoded as base64, whereas the latter is not possible to enforce signed integers in YAML.

Reply:
Bytes. Can you specify how exactly to specify them? E.g. the following is base64-encoded, but parses to a string, not a byte array:

```
cargo run --bin minimal_subscriber -- --ros-args --param foo:="eW9sbw=="
```

Reply:
I meant that they can be encoded, but there's no way syntactically to differentiate bytes from strings.